Code/Framework/AzFramework/AzFramework/Windowing/NativeWindow.cpp

```diff
@@ -90,82 +97,170 @@ namespace AzFramework
+        if (m_pimpl != nullptr)
+        {
+            return m_pimpl->IsActive();
+        }
+        else
+        {
+            return false;
+        }
     }
 
     void NativeWindow::SetWindowTitle(const AZStd::string& title)
     {
-        m_pimpl->SetWindowTitle(title);
+        if (m_pimpl != nullptr)
+        {
+            m_pimpl->SetWindowTitle(title);
+        }
     }
 
     WindowSize NativeWindow::GetClientAreaSize() const
     {
-        return m_pimpl->GetClientAreaSize();
+        if (m_pimpl != nullptr)
+        {
+            return m_pimpl->GetClientAreaSize();
+        }
+        else
+        {
+            return WindowSize();
+        }
     }
 
     WindowSize NativeWindow::GetMaximumClientAreaSize() const
     {
-        return m_pimpl->GetMaximumClientAreaSize();
+        if (m_pimpl != nullptr)
+        {
+            return m_pimpl->GetMaximumClientAreaSize();
+        }
+        else
+        {
+            return WindowSize();
+        }
     }
 
     void NativeWindow::ResizeClientArea(WindowSize clientAreaSize, const WindowPosOptions& options)
     {
-        m_pimpl->ResizeClientArea(clientAreaSize, options);
+        if (m_pimpl != nullptr)
+        {
+            m_pimpl->ResizeClientArea(clientAreaSize, options);
+        }
     }
 
     bool NativeWindow::SupportsClientAreaResize() const
     {
-        return m_pimpl->SupportsClientAreaResize();
+        if (m_pimpl != nullptr)
+        {
+            return m_pimpl->SupportsClientAreaResize();
+        }
+        else
+        {
+            return false;
+        }
     }
 
     void NativeWindow::SetEnableCustomizedResolution(bool enable)
     {
-        m_pimpl->SetEnableCustomizedResolution(enable);
+        if (m_pimpl != nullptr)
+        {
+            m_pimpl->SetEnableCustomizedResolution(enable);
+        }
     }
 
     bool NativeWindow::IsCustomizedResolutionEnabled() const
     {
-        return m_pimpl->IsCustomizedResolutionEnabled();
+        if (m_pimpl != nullptr)
+        {
+            return m_pimpl->IsCustomizedResolutionEnabled();
+        }
+        else
+        {
+            return false;
+        }
     }
 
     WindowSize NativeWindow::GetRenderResolution() const
     {
-        return m_pimpl->GetRenderResolution();
+        if (m_pimpl != nullptr)
+        {
+            return m_pimpl->GetRenderResolution();
+        }
+        else
+        {
+            return WindowSize();
+        }
     }
 
     void NativeWindow::SetRenderResolution(WindowSize resolution)
     {
-        m_pimpl->SetRenderResolution(resolution);
+        if (m_pimpl != nullptr)
+        {
+            m_pimpl->SetRenderResolution(resolution);
+        }
     }
 
     bool NativeWindow::GetFullScreenState() const
     {
-        return m_pimpl->GetFullScreenState();
+        if (m_pimpl)
+        {
+            return m_pimpl->GetFullScreenState();
+        }
+        else
+        {
+            return false;
+        }
```

Comment:
repeat: This can be made to a logical-AND return statement
```suggestion
        return m_pimpl != nullptr && m_pimpl->GetFullScreenState();
```